rust/protocol/src/sealed_sender.rs

```diff
@@ -879,8 +1019,214 @@ mod sealed_sender_v2 {
+///     message = AEAD_Decrypt(K, ciphertext) // includes S.public
+///     AT' = KDF(label_DH_s, DH(S, R) || E.public || C || S.public || R.public, len=16)
+///     if AT != AT':
+///         return DecryptionError
+///     return message
+///```
+///
+/// # Wire Format
+/// Multi-recipient sealed-sender does not use protobufs for its payload format. Instead, it uses
+/// a flat format marked with a [version byte](#the-version-byte). The format is different for
+/// [sending] and [receiving]. The decrypted content is
+/// a protobuf-encoded `UnidentifiedSenderMessage.Message` from `sealed_sender.proto`.
+///
+/// The public key used in Sealed Sender v2 is always a Curve25519 DJB key.
+///
+/// [sending]: #sent-messages
+/// [receiving]: #received-messages
+///
+/// ## The version byte
+///
+/// Sealed sender messages (v1 and v2) in serialized form begin with a version [byte][u8].
+/// This byte has the form:
+///
+/// ```text
+/// (requiredVersion << 4) | currentVersion
+/// ```
+///
+/// v1 messages thus have a version byte of `0x11`. v2 messages have a version byte
+/// of `0x22`. A hypothetical version byte `0x34` would indicate a message encoded
+/// as Sealed Sender v4, but decodable by any client that supports Sealed Sender v3.
+///
+/// ## Routing messages to recipients
+///
+/// The Signal backend server uses the initial version byte from the [sent message][sending] to
+/// determine whether the message was sent using Sealed Sender v2. If so, the server will then split
+/// up the set of messages and securely route each individual [received message][receiving] to its
+/// intended recipient.
+///
+/// For testing purposes, [sealed_sender_multi_recipient_fan_out] can be used to convert such a bulk
+/// message produced by Sealed Sender v2 into a sequence of [received messages][receiving]; however,
+/// in doing so it will drop all of the metadata necessary to identify the message's
+/// intended recipient.
+///
+/// ## Received messages
+///
+/// ```text
+/// ReceivedMessage {
+///     version_byte: u8,
+///     c: [u8; 32],
+///     at: [u8; 16],
+///     e_pub: [u8; 32],
+///     message: [u8] // remaining bytes
+/// }
+/// ```
+///
+/// Each individual Sealed Sender message received from the server is decoded in the Signal
+/// client by calling [sealed_sender_decrypt].
+///
+/// ## Sent messages
+///
+/// ```text
+/// PerRecipientData {
+///     uuid: [u8; 16],
+///     device_id: varint,
+///     registration_id: u16,
+///     c: [u8; 32],
+///     at: [u8; 16],
+/// }
+///
+/// SentMessage {
+///     version_byte: u8,
+///     count: varint,
+///     recipients: [PerRecipientData; count],
+///     e_pub: [u8; 32],
+///     message: [u8] // remaining bytes
+/// }
+/// ```
+///
+/// The varint encoding used is the same as [protobuf's][varint]. Values are unsigned. UUIDs are
+/// encoded per [RFC 4122], with the first eight bytes considered "most significant" [^1].
+/// Fixed-width integers are in network byte order (big-endian).
```

Comment:
Thought of something I forgot to put in the original doc: "Fixed-width integers are unaligned and in network byte order".

Reply:
Made this change!